### PR TITLE
Add node version check

### PR DESCRIPTION
**Pull request by peripateticus**
- Adding a node version check at ~`preinstall`~ `postinstall` that will error if the client version is less than the version specified by `engines.node`.

Error Message:
Screen Shot 2019-08-21 at 10 31 09 AM

Also logging current version at build time:
```
info: [febs] Current node version: v10.15.1. <---
info: [febs] Log Level: info
info: [febs] Building in webpack production mode...
...
```

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1 +1,31 @@
-module.exports = {};
+module.exports = {
+  /**
+   * Check that client is on minimum supported node version located.
+   * @param {String} currentVersion The current node version as returned by node, e.g. v10.15.1.
+   * @param {String} minVersion The minimum node version supported (engines.node)
+   * @returns {boolean|*} Whether or not client node version is greater than minVersion.
+   */
+  checkVersion: (currentVersion, minVersion) => {
```
Comment (dewilli-rei) [inline lib/index.js]: Could we make this less complicated and more robust by using the semver dependency?

We already trust the semver parser for node (the one npm uses) https://github.com/npm/node-semver

**Comment by peripateticus** [comment]
> Could we make this less complicated and more robust by using the semver dependency?
> 
> We already trust the semver parser for node (the one npm uses) https://github.com/npm/node-semver

We could for sure. I was initially considering but honestly I didn't think this functionality really warranted pulling in all of `semver`'s functionality, [1600 lines](https://github.com/npm/node-semver/blob/master/semver.js#L1596) of code. My function is like [20 lines](https://github.com/rei/febs/blob/version-check/lib/index.js#L9-L29)?

**Comment by peripateticus** [comment]
I pushed up some unrelated commits that fix and improve the coverage reporting. Currently, coverage reports are just being generated individually on each run but the generated report written to the file system is just the last run. This update will consolidate all of the results into a single report:

`$ npm run cover` ↴
Screen Shot 2019-08-22 at 8 01 15 PM

HTML report ↴
Screen Shot 2019-08-22 at 8 11 40 PM

File: bin/preinstall.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+const fs = require('fs');
+const path = require('path');
+const lib = require('../lib');
+
+// Version check for node.js. Should be >= engines.node.
+const febsPackageJson = fs.readFileSync(path.join(__dirname, '..', 'package.json'));
+const enginesNodeVersion = JSON.parse(febsPackageJson.toString()).engines.node;
+const minNodeVersion = enginesNodeVersion.split('').filter(c => Number.parseInt(c, 10) >= 0).join('.');
+
+if (!lib.checkVersion(process.version, minNodeVersion)) {
```
Comment (zcrumbo) [inline bin/preinstall.js]: I'm on the fence about including `semver` here. It does seem like it would be useful in the future.

Comment (peripateticus) [reply bin/preinstall.js]: Thanks for the feedback @zcrumbo.  @cowills Where are you leaning?

**Comment by peripateticus** [comment]
OK switched us over to `semver`. I recall now the other reason why I initially didn't use `semver`. This code is executed on `preinstall` so I was *thinking* that any dependencies would not yet be installed. Apparently this is not the case, dependencies are installed and *then* `preinstall` is called.

**Comment by peripateticus** [comment]
> OK switched us over to `semver`. I recall now the other reason why I initially didn't use `semver`. This code is executed on `preinstall` so I was _thinking_ that any dependencies would not yet be installed. Apparently this is not the case, dependencies are installed and _then_ `preinstall` is called.

Scratch that, it wasn't that `@rei/febs` dependencies were installed, it was that `semver` was already installed via another client dependency.

**Comment by peripateticus** [comment]
OK, you see our dilemma now...if we want to use `semver` we can't call this code on `preinstall`. Next best thing,  `postinstall`, any objections?

**Comment by peripateticus** [comment]
OK, running on `postinstall` now:
```
$ npm install

> @rei/febs@5.3.0 postinstall /path/to/proj/node_modules/@rei/febs
> ./bin/postinstall.js

⛔ febs requires node.js >=8.3.0. You are on version v8.0.0
```

**Pull request merged**